dulwich/tests/test_repository.py

```diff
@@ -765,3 +765,8 @@ def test_commit_no_encode_decode(self):
             mode, id = tree_lookup_path(r.get_object, r[commit_sha].tree, name)
             self.assertEqual(stat.S_IFREG | 0o644, mode)
             self.assertEqual(encoding.encode('ascii'), r[id].data)
+
+    def test_discover(self):
```

Comment:
please add two more tests; one for opening from a path that doesn't have a git repo (e.g. "/") and one that has a repository in the path specified.